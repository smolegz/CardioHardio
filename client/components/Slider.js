import React from "react";
import { StyleSheet, Text, View, StatusBar, Image } from "react-native";
import AppIntroSlider from "react-native-app-intro-slider";
import { useNavigation } from "@react-navigation/native";

const data = [
  {
    title: "Welcome to CardioHARDio",
    text: "An easy-to-use app to keep you healthy!",
    image: require("../assets/Slider/slider1.png"),
  },
  {
    title: "Stay active with us!",
    text: "Our app offer features that you'll need daily.",
    image: require("../assets/Slider/slider2.png"),
  },
  {
    title: "Keeping a healthy heart!",
    text: "Stay healthy for your loved ones!",
    image: require("../assets/Slider/slider3.png"),
  },
];

const Slider = () => {
  const navigation = useNavigation();

  const renderItem = ({ item }) => {
    return (
      <View style={styles.slide}>
        <Image source={item.image} style={styles.image} />
        <View>
          <Text style={styles.title}>{item.title}</Text>
          <Text style={styles.text}>{item.text}</Text>
        </View>
      </View>
    );
  };

  const keyExtractor = (item) => item.title;

  const renderDoneButton = () => {
    return (
      <View style={styles.button}>
        <Text style={styles.buttonText}>Proceed</Text>
      </View>
    );
  };

  const onDone = () => {
    navigation.replace("Welcome Home");
  };

  return (
    <View style={{ flex: 1 }}>
      <StatusBar translucent backgroundColor="transparent" />
      <AppIntroSlider
        keyExtractor={keyExtractor}
        renderItem={renderItem}
        renderDoneButton={renderDoneButton}
        dotStyle={styles.dotStyle}
        activeDotStyle={styles.activeDotStyle}
        data={data}
        onDone={onDone}
      />
    </View>
  );
};

export default Slider;

const styles = StyleSheet.create({
  slide: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "white",
  },
  image: {
    height: "40%",
    width: "90%",
    borderRadius: 50,
    borderWidth: 1,
    borderColor: "transparent",
    marginBottom: 30,
  },
  title: {
    fontFamily: "FiraSans_700Bold",
    fontSize: 28,
    textAlign: "center",
    marginBottom: 10,
  },
  text: {
    fontFamily: "FiraSans_300Light",
    fontSize: 16,
    textAlign: "center",
  },
  button: {
    borderRadius: 20,
    padding: 10,
    paddingHorizontal: 20,
    shadowColor: "black",
    shadowOffset: { width: 2, height: 2 },
    shadowOpacity: 0.3,
    shadowRadius: 2,
    backgroundColor: "#212A3E",
    borderWidth: 1,
    borderColor: "#212A3E",
  },
  buttonText: {
    color: "white",
    fontFamily: "FiraSans_300Light",
  },
  dotStyle: {
    backgroundColor: "#E0E0E0",
  },
  activeDotStyle: {
    backgroundColor: "#212A3E",
  },
});
