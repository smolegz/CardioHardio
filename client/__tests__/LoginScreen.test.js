import React from "react";
import {create, act} from 'react-test-renderer';
import Slider from "../components/Slider";

let root;

describe("Slider", () => {
  it("renders correctly", () => {
    const tree = create(<Slider />).toJSON();
    expect(tree).toMatchSnapshot();
  });
})


