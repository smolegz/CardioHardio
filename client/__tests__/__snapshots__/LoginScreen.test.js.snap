// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      bottomButton={false}
      bounces={false}
      data={
        [
          {
            "image": 1,
            "text": "An easy-to-use app to keep you healthy!",
            "title": "Welcome to CardioHARDio",
          },
          {
            "image": 1,
            "text": "Our app offer features that you'll need daily.",
            "title": "Stay active with us!",
          },
          {
            "image": 1,
            "text": "Stay healthy for your loved ones!",
            "title": "Keeping a healthy heart!",
          },
        ]
      }
      dotClickEnabled={true}
      extraData={1}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={3}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onDone={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={false}
      renderDoneButton={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showDoneButton={true}
      showNextButton={true}
      showPrevButton={false}
      showSkipButton={false}
      showsHorizontalScrollIndicator={false}
      stickyHeaderIndices={[]}
      style={
        {
          "flex": 1,
          "flexDirection": "row",
        }
      }
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "flex": 1,
                "width": 0,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Image
                source={1}
                style={
                  {
                    "borderColor": "transparent",
                    "borderRadius": 50,
                    "borderWidth": 1,
                    "height": "40%",
                    "marginBottom": 30,
                    "width": "90%",
                  }
                }
              />
              <View>
                <Text
                  style={
                    {
                      "fontFamily": "FiraSans_700Bold",
                      "fontSize": 28,
                      "marginBottom": 10,
                      "textAlign": "center",
                    }
                  }
                >
                  Welcome to CardioHARDio
                </Text>
                <Text
                  style={
                    {
                      "fontFamily": "FiraSans_300Light",
                      "fontSize": 16,
                      "textAlign": "center",
                    }
                  }
                >
                  An easy-to-use app to keep you healthy!
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "flex": 1,
                "width": 0,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Image
                source={1}
                style={
                  {
                    "borderColor": "transparent",
                    "borderRadius": 50,
                    "borderWidth": 1,
                    "height": "40%",
                    "marginBottom": 30,
                    "width": "90%",
                  }
                }
              />
              <View>
                <Text
                  style={
                    {
                      "fontFamily": "FiraSans_700Bold",
                      "fontSize": 28,
                      "marginBottom": 10,
                      "textAlign": "center",
                    }
                  }
                >
                  Stay active with us!
                </Text>
                <Text
                  style={
                    {
                      "fontFamily": "FiraSans_300Light",
                      "fontSize": 16,
                      "textAlign": "center",
                    }
                  }
                >
                  Our app offer features that you'll need daily.
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "flex": 1,
                "width": 0,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Image
                source={1}
                style={
                  {
                    "borderColor": "transparent",
                    "borderRadius": 50,
                    "borderWidth": 1,
                    "height": "40%",
                    "marginBottom": 30,
                    "width": "90%",
                  }
                }
              />
              <View>
                <Text
                  style={
                    {
                      "fontFamily": "FiraSans_700Bold",
                      "fontSize": 28,
                      "marginBottom": 10,
                      "textAlign": "center",
                    }
                  }
                >
                  Keeping a healthy heart!
                </Text>
                <Text
                  style={
                    {
                      "fontFamily": "FiraSans_300Light",
                      "fontSize": 16,
                      "textAlign": "center",
                    }
                  }
                >
                  Stay healthy for your loved ones!
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        {
          "bottom": 16,
          "justifyContent": "center",
          "left": 16,
          "position": "absolute",
          "right": 16,
        }
      }
    >
      <RCTSafeAreaView>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 16,
              "justifyContent": "center",
              "margin": 16,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#212A3E",
                "borderRadius": 5,
                "height": 10,
                "marginHorizontal": 4,
                "opacity": 1,
                "width": 10,
              }
            }
          />
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#E0E0E0",
                "borderRadius": 5,
                "height": 10,
                "marginHorizontal": 4,
                "opacity": 1,
                "width": 10,
              }
            }
          />
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#E0E0E0",
                "borderRadius": 5,
                "height": 10,
                "marginHorizontal": 4,
                "opacity": 1,
                "width": 10,
              }
            }
          />
        </View>
        <View
          style={
            {
              "position": "absolute",
              "right": 0,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                {
                  "color": "white",
                  "fontSize": 18,
                  "padding": 12,
                }
              }
            >
              Next
            </Text>
          </View>
        </View>
      </RCTSafeAreaView>
    </View>
  </View>
</View>
`;
